### ISSUE TITLE
request: put technology specs in a config file

### ISSUE THREAD

**Issue by jamestwebber** [pachterlab/kallisto#190]
With the release of `kallisto bus` and `bustools` I've been exploring how we could use these tools to upgrade our current scRNAseq pipelines. One thing that struck me is that the supported single-cell technologies are hard-coded in `main.cpp`, but the format appears to be pretty standardized.

It seems like it would be easy to break that out into a config file of some sort that the user could access, which would make it easy to 1) inspect the technology specification to verify that it matches the experiment, and 2) add new specifications that might be non-standard. This could help people use kallisto when developing new sequencing protocols.

edit: as a sidenote, the install in `conda` claims to be `0.44` although I believe it's only missing the commit that bumped the version number...

**Comment by lakigigar** [comment]
Yes, we agree and @pmelsted has already been working on an option for passing the barcode/UMI format as an option. 

Also thanks for the note about the version; we forgot to update it before the release. This will be fixed with the next release.

**Comment by jamestwebber** [comment]
Whoops, wrong button.

Sounds like you folks are already on it, sounds good. The use-case I have in mind is to process our Smart-seq2 data with a similar pipeline, although the lack of UMIs means that the U in BUS doesn't apply. But we'll hopefully have UMIs in future Smart-seq2 datasets, so it'd still be useful.
